pkg/util/pki/parse.go

```diff
@@ -252,8 +252,14 @@ func (c *chainNode) toBundleAndCA() (PEMBundle, error) {
 
 	for {
 		// If the issuer is nil, we have hit the root of the chain. Assign the CA
-		// to this certificate and stop traversing.
+		// to this certificate and stop traversing. If the root certificate is not
+		// a self-signed certificate, i.e. is not a root CA, append the certificate
+		// to the certificate chain.
```

Comment:
nitpick: minor suggestion here to clarify the difference between the root of the chain we've built and a "root certificate" more generally.

```suggestion
		// to this certificate and stop traversing. If the certificate at the root of
		// the chain is not self-signed (i.e. is not a root CA), then append the
		// certificate to the certificate chain.
```